search/includes/classes/class-settingshealthjob.php

```diff
@@ -149,24 +162,31 @@ public function heal_index_settings( $unhealthy_indexables ) {
 
 			// Each individual entry in $versions is an array of results, one per index version.
 			foreach ( $versions as $result ) {
-				// Only take action if there are actual inconsistencies in auto-healable keys.
-				$diff = $this->health::limit_index_settings_to_keys( $result['diff'], $this->health::INDEX_SETTINGS_HEALTH_AUTO_HEAL_KEYS );
-				if ( empty( $diff ) ) {
-						continue;
+				if ( empty( $result['diff'] ) ) {
+					continue;
 				}
+				// Check if index needs to be re-built in the background or just do auto-healing.
+				if ( true === array_key_exists( 'index.number_of_shards', $result['diff'] ) ) {
+					$this->maybe_build_new_index( $indexable );
+				} else {
+					$diff = $this->health::limit_index_settings_to_keys( $result['diff'], $this->health::INDEX_SETTINGS_HEALTH_AUTO_HEAL_KEYS );
+					if ( empty( $diff ) ) {
+						continue;
+					}
```

Comment:
This will stop the auto-healing I think if shards count is a mismatch, is that intended?

So if a site ends up "stuck" in the position where it has the wrong number of shards but can't auto-create a new one (already has 2+ indexes) - it will also stop auto-healing the one it is using.

Reply:
@WPprodigy It was intentional because it doesn't make sense to update the index settings if the index is being re-built (whether automatically or manually after alerting). 

Good point though — might be worth it to do it anyways to prevent a scenario where a site with 2+ indexes does not get the autohealing of the other index settings.